src/common/include/SIRF/common/geometrical_info.h

```diff
@@ -0,0 +1,47 @@
+#ifndef SIRF_GEOMETRICAL_INFO_TYPE
+#define SIRF_GEOMETRICAL_INFO_TYPE
+
+const int DIM = 3;
+
+template <int num_dimensions, typename CoordT>
+class Coordinate {
+	typedef CoordT _CoordsT[num_dimensions];
+
+public:
+	inline CoordT& operator[](const int d)
+		{ return coords[d]; }
+	inline CoordT operator[](const int d) const
+		{ return coords[d]; }
+
+private:
+	_CoordsT coords;
+};
+
+class GeometricalInfo {
+public:
+	typedef Coordinate<DIM, float>                  Offset;
+	typedef Coordinate<DIM, float>                  Spacing;
+	typedef Coordinate<DIM, int>                    Size;
+	typedef Coordinate<DIM, Coordinate<DIM, float>> Direction;
+
+
+	GeometricalInfo(Offset offset, Spacing spacing, Size size, Direction direction);
+	GeometricalInfo(const GeometricalInfo& other) {
+		*this = other;
+	}
+	~GeometricalInfo() {};
```

Comment:
either delete or (better) make virtual (all base classes should really have a virtual destructor)

Reply:
Thanks. A Google inquiring why lead to a useful lesson on the virtual keyword.